File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/wallet/OpenWalletTest.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.IndyIntegrationTest;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertNotNull;

import org.json.JSONObject;
import org.junit.Test;

import java.util.concurrent.ExecutionException;


public class OpenWalletTest extends IndyIntegrationTest {

	@Test
	public void testOpenWalletWorks() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		Wallet wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		assertNotNull(wallet);

		wallet.closeWallet().get();
	}

	@Test
	public void testOpenWalletWorksForInvalidCredentials() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletAccessFailedException.class));

		Wallet.openWallet(WALLET_CONFIG, "{\"key\": \"other_key\"}").get();
	}

	@Test
	public void testOpenWalletWorksForChangingCredentials() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, "{\"key\": \"key\"}").get();

		Wallet wallet = Wallet.openWallet(WALLET_CONFIG, "{\"key\": \"key\", \"rekey\": \"other_key\"}").get();
		wallet.closeWallet().get();

		wallet = Wallet.openWallet(WALLET_CONFIG, "{\"key\": \"other_key\"}").get();
		wallet.closeWallet().get();
	}

	@Test
	public void testOpenWalletWorksForNotCreatedWallet() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletNotFoundException.class));

		Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testOpenWalletWorksForTwice() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletAlreadyOpenedException.class));

		String config = new JSONObject()
				.put("id", "openWalletWorksForTwice")
				.toString();

		Wallet.createWallet(config, WALLET_CREDENTIALS).get();

		Wallet.openWallet(config, WALLET_CREDENTIALS).get();
		Wallet.openWallet(config, WALLET_CREDENTIALS).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/OpenWalletTest.cs
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class OpenWalletTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public async Task TestOpenWalletWorks()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            Assert.IsNotNull(wallet);
            await wallet.CloseAsync();
        }

        [TestMethod]
        public async Task TestOpenWalletWorksForForInvalidCredentials()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            var ex = await Assert.ThrowsExceptionAsync<WalletAccessFailedException>(() =>
                Wallet.OpenWalletAsync(WALLET_CONFIG, "{\"key\": \"other_key\"}")
            );
        }

        [TestMethod]
        public async Task TestOpenWalletWorksForChangingCredentials()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, "{\"key\": \"key\"}");
            
            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, "{\"key\": \"key\", \"rekey\": \"other_key\"}");
            await wallet.CloseAsync();

            wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, "{\"key\": \"other_key\"}");
            await wallet.CloseAsync();
        }
        
        [TestMethod]
        public async Task TestOpenWalletWorksForNotCreated()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletNotFoundException>(() =>
               Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS)
            );
        }
        
        [TestMethod]
        public async Task TestOpenWalletWorksForTwice()
        {
            var config = JsonConvert.SerializeObject(new { id = "openWalletWorksForTwice" });

            await Wallet.CreateWalletAsync(config, WALLET_CREDENTIALS);
            var wallet = await Wallet.OpenWalletAsync(config, WALLET_CREDENTIALS);

            var ex = await Assert.ThrowsExceptionAsync<WalletAlreadyOpenedException>(() =>
               Wallet.OpenWalletAsync(config, WALLET_CREDENTIALS)
            );

            await wallet.CloseAsync();
        }        
    }
}
